ensure_command_output.sh

```diff
@@ -1,59 +1,43 @@
 #!/bin/bash
 
-MAGIC_END_STRING="END OF SUBCOMMAND"
 OUTPUT_SLEEP=60
 
 function run_command {
-  PIPE=$1
-  shift
   CMD="$@"
-  $CMD 1>$PIPE 2>$PIPE
-  echo "$MAGIC_END_STRING" 1>$PIPE
+  $CMD 
 }
 
 function run_ping {
-  PIPE=$1
-  shift
   CMD="$@"
-  (
     while [ 1 ]
     do
       echo "Command '$CMD' is still running."
       sleep $OUTPUT_SLEEP
     done
-  ) >$PIPE
 }
 
 function presto_wait {
-  PIPE=$1
-  shift
   CMD="$@"
 
-  mkfifo $PIPE
-  echo "Starting command '$CMD' using pipe $PIPE"
-  run_command $PIPE $CMD &
-  run_ping $PIPE $CMD &
+  echo "Starting command '$CMD'"
+  run_command $CMD &
```

Comment:
Do you really need to wrap it with a method? Sending bash function to background causes bash fork. And you will get `pid` not of the command itself, but of the bash wrapper. I would rather un-wrap it.